trackintel/io/postgis.py

```diff
@@ -22,348 +56,259 @@ def read_positionfixes_postgis(conn_string, table_name, geom_col="geom", *args,
 ):
-    """Stores positionfixes to PostGIS. Usually, this is directly called on a positionfixes
-    DataFrame (see example below).
+    positionfixes.to_postgis(name, con, schema, if_exists, index, index_label, chunksize, dtype)
 
-    Parameters
-    ----------
-    positionfixes : GeoDataFrame
-        The positionfixes to store to the database.
 
-    conn_string : str
-        A connection string to connect to a database, e.g.,
-        ``postgresql://username:password@host:socket/database``.
-
-    table_name : str
-        The name of the table to write to.
-
-    schema : str, optional
-        The schema (if the database supports this) where the table resides.
-
-    sql_chunksize : int, optional
-        How many entries should be written at the same time.
-
-    if_exists : str, {'fail', 'replace', 'append'}, default 'fail'
-        What should happen if the table already exists.
-
-    Examples
-    --------
-    >>> df.as_positionfixes.to_postgis(conn_string, table_name)
-    """
-    engine = create_engine(conn_string)
-    conn = engine.connect()
-    try:
-        positionfixes.to_postgis(table_name, conn, if_exists=if_exists, index=True, chunksize=sql_chunksize)
-    finally:
-        conn.close()
-
-
-def read_triplegs_postgis(conn_string, table_name, geom_col="geom", *args, **kwargs):
+@_handle_con_string
+def read_triplegs_postgis(sql, con, geom_col="geom", *args, **kwargs):
     """Reads triplegs from a PostGIS database.
 
     Parameters
     ----------
-    conn_string : str
-        A connection string to connect to a database, e.g.,
-        ``postgresql://username:password@host:socket/database``.
+    sql : str
+        SQL query e.g. "SELECT * FROM triplegs"
 
-    table_name : str
-        The table to read the triplegs from.
+    con : str, sqlalchemy.engine.Connection or sqlalchemy.engine.Engine
+        Connection string or active connection to PostGIS database.
 
     geom_col : str, default 'geom'
         The geometry column of the table.
 
+    *args
+        Further arguments as available in GeoPanda's GeoDataFrame.from_postgis().
+
+    **kwargs
+        Further keyword arguments as available in GeoPanda's GeoDataFrame.from_postgis().
+
     Returns
     -------
     GeoDataFrame
         A GeoDataFrame containing the triplegs.
-    """
-    engine = create_engine(conn_string)
-    conn = engine.connect()
-    try:
-        pfs = gpd.GeoDataFrame.from_postgis(
-            "SELECT * FROM %s" % table_name, conn, geom_col=geom_col, index_col="id", *args, **kwargs
-        )
-    finally:
-        conn.close()
 
-    # assert validity of triplegs
-    pfs.as_triplegs
-    return pfs
+    Examples
+    --------
+    >>> tpls = ti.io.read_triplegs_postgis("SELECT * FROM triplegs", con, geom_col="geom")
+    """
+    tpls = gpd.GeoDataFrame.from_postgis(sql, con, geom_col=geom_col, index_col="id", *args, **kwargs)
```

Comment:
It is not allowed to use positional arguments (*args) after a keyword argument (index_col="id")